File: src/compiler/a5/grammar/PstToAstTokenVisitor.java
package compiler.a5.grammar;

import compiler.lexer.token.*;
import compiler.parser.visitors.TokenVisitor;

import static compiler.parser.PstToAstHelpers.hoist;

public class PstToAstTokenVisitor implements TokenVisitor<Void> {
  @Override
  public Void visit(SymbolToken.ForwardSlash forwardSlash) {
    hoist(forwardSlash);
    return null;
  }

  @Override
  public Void visit(OperatorToken.Asterisk asterisk) {
    hoist(asterisk);
    return null;
  }

  @Override
  public Void visit(SymbolToken.LeftParen token) {
    return null;
  }

  @Override
  public Void visit(IdentifierToken token) {
    return null;
  }

  @Override
  public Void visit(CommentToken token) {
    return null;
  }

  @Override
  public Void visit(EOFToken token) {
    return null;
  }

  @Override
  public Void visit(FloatToken token) {
    return null;
  }

  @Override
  public Void visit(IntegerToken token) {
    return null;
  }

  @Override
  public Void visit(KeywordToken.MainKeywordToken token) {
    return null;
  }

  @Override
  public Void visit(KeywordToken.FunctionKeywordToken token) {
    return null;
  }

  @Override
  public Void visit(TypeToken.FloatKeywordToken token) {
    return null;
  }

  @Override
  public Void visit(TypeToken.IntegerKeywordToken token) {
    return null;
  }

  @Override
  public Void visit(TypeToken.StringKeywordToken token) {
    return null;
  }

  @Override
  public Void visit(KeywordToken.IfKeywordToken token) {
    return null;
  }

  @Override
  public Void visit(KeywordToken.ElseIfKeywordToken token) {
    return null;
  }

  @Override
  public Void visit(KeywordToken.ElseKeywordToken token) {
    return null;
  }

  @Override
  public Void visit(KeywordToken.WhileKeywordToken token) {
    return null;
  }

  @Override
  public Void visit(KeywordToken.InputKeywordToken token) {
    return null;
  }

  @Override
  public Void visit(KeywordToken.PrintKeywordToken token) {
    return null;
  }

  @Override
  public Void visit(KeywordToken.NewKeywordToken token) {
    return null;
  }

  @Override
  public Void visit(KeywordToken.ReturnKeywordToken token) {
    return null;
  }

  @Override
  public Void visit(OperatorToken.LessThan token) {
    return null;
  }

  @Override
  public Void visit(OperatorToken.GreaterThan token) {
    return null;
  }

  @Override
  public Void visit(OperatorToken.Equal token) {
    return null;
  }

  @Override
  public Void visit(OperatorToken.Minus token) {
    return null;
  }

  @Override
  public Void visit(OperatorToken.Plus token) {
    return null;
  }

  @Override
  public Void visit(OperatorToken.Ampersand token) {
    return null;
  }

  @Override
  public Void visit(OperatorToken.Arrow token) {
    return null;
  }

  @Override
  public Void visit(OperatorToken.EqualEqual token) {
    return null;
  }

  @Override
  public Void visit(OperatorToken.NotEqual token) {
    return null;
  }

  @Override
  public Void visit(OperatorToken.LessThanOrEqual token) {
    return null;
  }

  @Override
  public Void visit(OperatorToken.GreaterThanOrEqual token) {
    return null;
  }

  @Override
  public Void visit(OperatorToken.BitShiftLeft token) {
    return null;
  }

  @Override
  public Void visit(OperatorToken.BitShiftRight token) {
    return null;
  }

  @Override
  public Void visit(StringToken token) {
    return null;
  }

  @Override
  public Void visit(SymbolToken.Comma token) {
    return null;
  }

  @Override
  public Void visit(SymbolToken.SemiColon token) {
    return null;
  }

  @Override
  public Void visit(SymbolToken.LeftBrace token) {
    return null;
  }

  @Override
  public Void visit(SymbolToken.RightBrace token) {
    return null;
  }

  @Override
  public Void visit(SymbolToken.LeftBracket token) {
    return null;
  }

  @Override
  public Void visit(SymbolToken.RightBracket token) {
    return null;
  }

  @Override
  public Void visit(SymbolToken.RightParen token) {
    return null;
  }

  @Override
  public Void visit(SymbolToken.Caret token) {
    return null;
  }

  @Override
  public Void visit(SymbolToken.Colon token) {
    return null;
  }

  @Override
  public Void visit(SymbolToken.Period token) {
    return null;
  }

  @Override
  public Void visit(WhitespaceToken token) {
    return null;
  }

  @Override
  public Void visit(KeywordToken.ClassKeywordToken token) {
    return null;
  }

  @Override
  public Void visit(KeywordToken.ProgramKeywordToken token) {
    return null;
  }

  @Override
  public Void visit(KeywordToken.VarKeywordToken token) {
    return null;
  }

}

File: src/compiler/a5/grammar/PstToAstTokenVisitor.kt
package compiler.a5.grammar

import compiler.parser.PstToAstHelpers
import compiler.parser.Symbol.Terminal.Ignorable.*
import compiler.parser.Symbol.Terminal.Keyword.*
import compiler.parser.Symbol.Terminal.Keyword.Type.*
import compiler.parser.Symbol.Terminal.Operator.*
import compiler.parser.Symbol.Terminal.Punctuation.*
import compiler.parser.Symbol.Terminal.TypedTerminal.*
import compiler.parser.visitors.TokenVisitor

class PstToAstTokenVisitor : TokenVisitor<Unit> {
    override fun visit(token: ForwardSlash) {
        PstToAstHelpers.hoist(token)
    }

    override fun visit(token: Asterisk) {
        PstToAstHelpers.hoist(token)
    }

    override fun visit(token: LeftParen) {
    }

    override fun visit(token: IdentifierTerminal) {
    }

    override fun visit(token: CommentTerminal) {
    }

    override fun visit(token: EOFTerminal) {
    }

    override fun visit(token: FloatTerminal) {
    }

    override fun visit(token: IntegerTerminal) {
    }

    override fun visit(token: MainKeyword) {
    }

    override fun visit(token: FunctionKeyword) {
    }

    override fun visit(token: FloatKeyword) {
    }

    override fun visit(token: IntegerKeyword) {
    }

    override fun visit(token: StringKeyword) {
    }

    override fun visit(token: IfKeyword) {
    }

    override fun visit(token: ElseIfKeyword) {
    }

    override fun visit(token: ElseKeyword) {
    }

    override fun visit(token: WhileKeyword) {
    }

    override fun visit(token: InputKeyword) {
    }

    override fun visit(token: PrintKeyword) {
    }

    override fun visit(token: NewKeyword) {
    }

    override fun visit(token: ReturnKeyword) {
    }

    override fun visit(token: LessThan) {
    }

    override fun visit(token: GreaterThan) {
    }

    override fun visit(token: Equal) {
    }

    override fun visit(token: Minus) {
    }

    override fun visit(token: Plus) {
    }

    override fun visit(token: Ampersand) {
    }

    override fun visit(token: Arrow) {
    }

    override fun visit(token: EqualEqual) {
    }

    override fun visit(token: NotEqual) {
    }

    override fun visit(token: LessThanOrEqual) {
    }

    override fun visit(token: GreaterThanOrEqual) {
    }

    override fun visit(token: BitShiftLeft) {
    }

    override fun visit(token: BitShiftRight) {
    }

    override fun visit(token: StringTerminal) {
    }

    override fun visit(token: Comma) {
    }

    override fun visit(token: SemiColon) {
    }

    override fun visit(token: LeftBrace) {
    }

    override fun visit(token: RightBrace) {
    }

    override fun visit(token: LeftBracket) {
    }

    override fun visit(token: RightBracket) {
    }

    override fun visit(token: RightParen) {
    }

    override fun visit(token: Caret) {
    }

    override fun visit(token: Colon) {
    }

    override fun visit(token: Period) {
    }

    override fun visit(token: WhitespaceTerminal) {
    }

    override fun visit(token: ClassKeyword) {
    }

    override fun visit(token: ProgramKeyword) {
    }

    override fun visit(token: VarKeyword) {
    }

    override fun visit(token: Void) {
    }

}
